ngraph/core/reference/include/ngraph/runtime/reference/matmul.hpp

```diff
@@ -86,43 +119,28 @@ namespace ngraph
                 const T* arg1_update = arg1;
 
                 // vars for updated inputs shapes
-                Shape wip_arg0_shape = arg0_shape;
-                Shape wip_arg1_shape = arg1_shape;
+                Shape arg0_shape_tmp = arg0_shape;
```

Comment:
Please check the whole function and add 'const' wherever is possible. It makes reading code much simpler if you know upfront if variable is mutable or not.

Reply:
Actually, these particular `_tmp` shapes can be modified with transpose but I've added missed `const` to other variables.